src/app/GitUI/UserControls/RevisionGrid/IndexWatcher.cs

```diff
@@ -89,8 +89,14 @@ private bool IndexChanged
             set
             {
                 _indexChanged = value;
-                GitIndexWatcher.EnableRaisingEvents = !IndexChanged
-                    && Directory.Exists(GitIndexWatcher.Path);
+                try
+                {
+                    GitIndexWatcher.EnableRaisingEvents = !IndexChanged
+                        && Directory.Exists(GitIndexWatcher.Path);
+                }
+                catch
+                {
```

Comment:
```suggestion
                catch
                {
                    GitIndexWatcher.EnableRaisingEvents = false;
```

Reply:
If thrown, the value is not set so this wold just be a documentation. Necessary?

Reply:
What if switching from normal wdir to WSL?

Reply:
If dir is changed, all these watchers are changed. You will always try to watch the current repo.

Reply:
> > > If thrown, the value is not set so this wold just be a documentation. Necessary?

No, not just documentation. The previous values remains. That's why I asked.

>> What if switching from normal wdir to WSL?

Think of a not-throwing implementation of `Directory.Exists`! It would return `false` in case of an invalid path.

> If dir is changed, all these watchers are changed. You will always try to watch the current repo.

Perhaps, the state machines might ensure that everything is reset. Though this is not obvious.